server/workflow_template_server.go

```diff
@@ -0,0 +1,334 @@
+package server
+
+import (
+	"context"
+	"errors"
+	"github.com/onepanelio/core/api"
+	v1 "github.com/onepanelio/core/pkg"
+	"github.com/onepanelio/core/pkg/util/label"
+	"github.com/onepanelio/core/server/auth"
+	"github.com/onepanelio/core/server/converter"
+	log "github.com/sirupsen/logrus"
+	"strings"
+	"time"
+)
+
+type WorkflowTemplateServer struct{}
+
+func NewWorkflowTemplateServer() *WorkflowTemplateServer {
+	return &WorkflowTemplateServer{}
+}
+
+func apiWorkflowTemplate(wft *v1.WorkflowTemplate) *api.WorkflowTemplate {
+	manifest, err := v1.RemoveAllButSpec(wft.GetManifestBytes())
+	if err != nil {
+		log.WithFields(log.Fields{
+			"Method": "apiWorkflowTemplate",
+			"Step":   "RemoveAllButSpec",
+			"Error":  err.Error(),
+		}).Error("Get Workflow Template failed.")
+	}
+
+	return &api.WorkflowTemplate{
+		Uid:        wft.UID,
+		CreatedAt:  wft.CreatedAt.UTC().Format(time.RFC3339),
+		Name:       wft.Name,
+		Version:    wft.Version,
+		Manifest:   string(manifest),
+		IsLatest:   wft.IsLatest,
+		IsArchived: wft.IsArchived,
+	}
+}
+
+func mapToKeyValue(input map[string]string) []*api.KeyValue {
+	var result []*api.KeyValue
+	for key, value := range input {
+		keyValue := &api.KeyValue{
+			Key:   key,
+			Value: value,
+		}
+
+		result = append(result, keyValue)
+	}
+
+	return result
+}
+
+func (s *WorkflowTemplateServer) CreateWorkflowTemplate(ctx context.Context, req *api.CreateWorkflowTemplateRequest) (*api.WorkflowTemplate, error) {
+	client := ctx.Value("kubeClient").(*v1.Client)
+	allowed, err := auth.IsAuthorized(client, req.Namespace, "create", "argoproj.io", "workflows", "")
```

Comment:
Now that we are using `WorkflowTemplates`, this should check for `workflowtemplates` instead of `workflows`:
```
auth.IsAuthorized(client, req.Namespace, "create", "argoproj.io", "workflowtemplates", "")
```

Reply:
This applies to all the other `auth.IsAuthorized` calls in this file.

Reply:
Should be fixed.